adr/ADR-31.md

```diff
@@ -104,18 +104,46 @@ Using the `batch` and `max_bytes` keys one can request multiple messages in a si
 
 The server will send multiple messages without any flow control to the reply subject, it will send up to `max_bytes` messages.  When `max_bytes` is unset the server will use the `max_pending` configuration setting or the server default (currently 64MB)
 
-After the batch is sent a zero length payload message will be sent with the `Nats-Num-Pending` and `Nats-Last-Sequence` headers set that clients can use to determine if further batch calls are needed. Additionally the `Nats-Last-Sequence` will hold the sequence of the last message sent. It would also have the `Status` header set to `204` with the `Description` header being `EOB`.
+After the batch is sent a zero length payload message will be sent with the `Nats-Num-Pending` and `Nats-Last-Sequence` headers set that clients can use to determine if further batch calls are needed. It would also have the `Status` header set to `204` with the `Description` header being `EOB`.
 
 When requests are made against servers that do not support `batch` the first response will be received and nothing will follow. Old servers can be detected by the absence of the `Nats-Num-Pending` header in the first reply.
 
+#### Multiple Subject requests
+
+Multiple subjects can be requested in the same manner that a Batch can be requested. In this mode we support consistent point in time reads by allowing for a group of subjects to be read as they were at a point in time - assuming the stream holds enough historical data.
+
+To help inform proper use of this feature, vs just using a consumer, a multi subject request may only allow matching no more than 1024 subjects, any more will result in a `413` status reply.
+
+Using requests like `{"multi_last":["$KV.USERS.1234.>"]}` all the latest values for all subjects below that wildcard will be returned.
```

Comment:
Not the case - the 1024 is a limit of subjects that match.  It wont return up to 1024 it will error telling you this feature only support subject sets <=1024.

What it will do is normal batching behaviour ie. cut the batch short if you reach pending buffer sizes

Reply:
I'll revisit thanks - I also miss understood this thing at first so maybe some left overs

Reply:
This seems fine to me, in essence:

```
There is a 1024 limit

....details of different kinds of requests follow
```

So made a tiny update there but I think its fine, cant mention for any time this is described the limit applies - it always applies